build/local/README.md

```diff
@@ -0,0 +1,127 @@
+# Local builds
+
+This directory contains a set of scripts aimed at providing a simple way to build PMM locally.
+
+## Background
+
+Historically, PMM used to be built using Jenkins. This worked well for the team, but not for the community. The learning curve was, and still is, rather steep, and it is hard for folks, even internally, to contribute to.
+
+Therefore, we decided to make it possible to build PMM locally. This is a work in progress, but we are definitely committed to bring the developer experience to an acceptable level.
+
+The build process is mostly based on bash scripts, which control the build flow. This was an intentional decision early on, since every developer should have at least a basic command of bash. Apart from bash and a few other well-known utilitites like `curl` or `make`, it also uses Docker for environment isolation and caching.
+
+The build process is designed to run on Linux or MacOS. We believe it could be adapated to run on other types of operating systems with little to no modification.
+
+
+## Prerequisites
+
+Below is a list of prerequisites that are required to build PMM locally.
+
+- OS: Linux (tested on Oracle Linux 9.3, Ubuntu 22.04.3 LTS), MacOS (tested on Sequoia 15.1)
+- Docker 25.0.2+
+- Docker [buildx plugin](https://github.com/docker/buildx) 0.16.0+
+- make
+- bash
+- tar
+- git
+- curl
+
+Also, it is required to define an environment variable `GITHUB_API_TOKEN` with a valid GitHub Personal Access Token (PAT). This token is used to pull the changes from the `percona-lab/pmm-submodules` and other project repositories. The token must have the following permissions:
+
+- repo:status
+- public_repo
+- read:user
+
+Please note, that building some of the PMM internals, such as Grafana, requires at least 8GB of memory available to docker. The number of CPUs, however, does not matter that much.
+
+## How to use this script to build PMM
+
+1. Install the prerequisites
+2. Clone the `pmm` repository - `git clone git@github.com:percona/pmm`.
+3. Change directory to `pmm` - `cd pmm`.
+4. Run `./build --help` to display the script usage.
+5. Run `./build --init` to provision all dependent submodules using `https://github.com/percona-lab/pmm-submodules` repository.
+6. Run `./build` with parameters of your choice to build PMM v3.
```

Comment:
Just minor: build.sh otherwise my terminal complains about folder of same name.

Reply:
Sure, I will update the README appropriately.